src/components/Carousel/CarouselDot.tsx

```diff
@@ -0,0 +1,44 @@
+import classNames from 'classnames';
+import { useInView } from 'react-intersection-observer';
+
+type Props = {
+  isSelected: boolean;
+  onClick: () => void;
+  baseClassName: string;
+  scrollableContainerRef: React.RefObject<HTMLDivElement>;
+  onInViewChange: (inView: boolean) => void;
+  variant?: 'sm' | 'md';
+};
+
+export const CarouselDot = ({
+  isSelected,
+  onClick,
+  baseClassName,
+  scrollableContainerRef,
+  onInViewChange,
+  variant = 'md',
+}: Props) => {
+  const { ref } = useInView({
+    threshold: 0,
+    root: scrollableContainerRef.current,
+    onChange(inView) {
+      onInViewChange(inView);
+    },
+  });
+
+  return (
+    <button
+      ref={ref}
+      role="button"
+      onClick={onClick}
+      className={classNames(`${baseClassName}-pagination-dot-container`)}
```

Comment:
BEM has a `__` for elements.

```suggestion
      className={classNames(`${baseClassName}-pagination-dot__container`)}
```